boa/src/syntax/lexer/tests.rs

```diff
@@ -264,270 +151,479 @@ fn check_keywords() {
+        TokenKind::Keyword(Keyword::With),
+        TokenKind::Keyword(Keyword::Yield),
+    ];
+
+    expect_tokens(&mut lexer, &expected);
 }
 
 #[test]
 fn check_variable_definition_tokens() {
     let s = "let a = 'hello';";
-    let mut lexer = Lexer::new(s);
-    lexer.lex().expect("failed to lex");
-    assert_eq!(lexer.tokens[0].kind, TokenKind::Keyword(Keyword::Let));
-    assert_eq!(lexer.tokens[1].kind, TokenKind::identifier("a"));
-    assert_eq!(
-        lexer.tokens[2].kind,
-        TokenKind::Punctuator(Punctuator::Assign)
-    );
-    assert_eq!(lexer.tokens[3].kind, TokenKind::string_literal("hello"));
+    let mut lexer = Lexer::new(s.as_bytes());
+
+    let expected = [
+        TokenKind::Keyword(Keyword::Let),
+        TokenKind::identifier("a"),
+        TokenKind::Punctuator(Punctuator::Assign),
+        TokenKind::string_literal("hello"),
+        TokenKind::Punctuator(Punctuator::Semicolon),
+    ];
+
+    expect_tokens(&mut lexer, &expected);
 }
 
 #[test]
 fn check_positions() {
-    let s = r#"console.log("hello world\u{2764}"); // Test"#;
+    let s = r#"console.log("hello world"); // Test"#;
     // --------123456789
-    let mut lexer = Lexer::new(s);
-    lexer.lex().expect("failed to lex");
+    let mut lexer = Lexer::new(s.as_bytes());
+
     // The first column is 1 (not zero indexed)
-    assert_eq!(lexer.tokens[0].span(), span((1, 1), (1, 7)));
+    assert_eq!(lexer.next().unwrap().unwrap().span(), span((1, 1), (1, 8)));
 
     // Dot Token starts on column 8
-    assert_eq!(lexer.tokens[1].span(), span((1, 8), (1, 8)));
+    assert_eq!(lexer.next().unwrap().unwrap().span(), span((1, 8), (1, 9)));
 
     // Log Token starts on column 9
-    assert_eq!(lexer.tokens[2].span(), span((1, 9), (1, 11)));
+    assert_eq!(lexer.next().unwrap().unwrap().span(), span((1, 9), (1, 12)));
 
     // Open parenthesis token starts on column 12
-    assert_eq!(lexer.tokens[3].span(), span((1, 12), (1, 12)));
+    assert_eq!(
+        lexer.next().unwrap().unwrap().span(),
+        span((1, 12), (1, 13))
+    );
 
     // String token starts on column 13
-    assert_eq!(lexer.tokens[4].span(), span((1, 13), (1, 33)));
+    assert_eq!(
+        lexer.next().unwrap().unwrap().span(),
+        span((1, 13), (1, 26))
+    );
 
-    // Close parenthesis token starts on column 34
-    assert_eq!(lexer.tokens[5].span(), span((1, 34), (1, 34)));
+    // Close parenthesis token starts on column 26.
+    assert_eq!(
+        lexer.next().unwrap().unwrap().span(),
+        span((1, 26), (1, 27))
+    );
 
     // Semi Colon token starts on column 35
-    assert_eq!(lexer.tokens[6].span(), span((1, 35), (1, 35)));
+    assert_eq!(
+        lexer.next().unwrap().unwrap().span(),
+        span((1, 27), (1, 28))
+    );
 }
 
 #[test]
-#[ignore]
-fn two_divisions_in_expression() {
-    let s = "    return a !== 0 || 1 / a === 1 / b;";
-    let mut lexer = Lexer::new(s);
-    lexer.lex().expect("failed to lex");
-    // dbg!(&lexer.tokens);
```

Comment:
I see that this was removed, but this was the main reason behind all this. This test is directly taken from test262. We should have it somewhere.

Reply:
It is, I moved it (and also split it into multiple tests so I could narrow down exactly the behaviour) - I'll go find them

Reply:
Moved to line 100 - 131 in boa/src/syntax/parser/tests.rs (since they involve the parser I thought they were better placed there).